core/controllers/custom_landing_pages.py

```diff
@@ -30,15 +30,27 @@ class FractionLandingRedirectPage(base.BaseHandler):
     @acl_decorators.open_access
     def get(self):
         """Handles GET requests."""
-        self.redirect('/learn/maths/fractions')
+        self.redirect('/maths/fractions')
+
+
+class TopicRedirectPage(base.BaseHandler):
+    """The handler redirecting the old landing page URL to the new one."""
+
+    @acl_decorators.open_access
+    def get(self, topic):
+        """Handles GET requests."""
+        self.redirect('/maths/%s' % topic)
 
 
 class TopicLandingPage(base.BaseHandler):
     """Page showing the topic landing page."""
 
     @acl_decorators.open_access
-    def get(self, subject, topic):
+    def get(self, topic):
         """Handles GET requests."""
+
+        subject = self.request.uri.split('/')[-2]
```

Comment:
What if the URL has fewer than 2 components? Is that a possibility?

Reply:
I don't think that's ever a possibility. Note that this is parsing the whole URI so even if this was to get `http://localhost:8181/` (which it won't since the get redirect function gives this handler only those URLs that are in the form of `/maths/<topic>`), it would still be alright since after splitting, the second last index would be `localhost:8181` which would not pass the valid topic check and a 404 response would be generated.